plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/resource/ResourcePayloadPage.java

```diff
@@ -0,0 +1,204 @@
+import org.eclipse.jface.dialogs.ErrorDialog;
+import org.eclipse.jface.dialogs.IDialogConstants;
+import org.eclipse.jface.layout.GridDataFactory;
+import org.eclipse.jface.layout.GridLayoutFactory;
+import org.eclipse.jface.wizard.IWizard;
+import org.eclipse.osgi.util.NLS;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.events.SelectionAdapter;
+import org.eclipse.swt.events.SelectionEvent;
+import org.eclipse.swt.events.SelectionListener;
+import org.eclipse.swt.widgets.Button;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.FileDialog;
+import org.eclipse.swt.widgets.Group;
+import org.eclipse.swt.widgets.Label;
+import org.eclipse.swt.widgets.Text;
+import org.eclipse.ui.dialogs.ElementTreeSelectionDialog;
+import org.jboss.tools.common.ui.databinding.ParametrizableWizardPageSupport;
+import org.jboss.tools.common.ui.databinding.SimpleUrlStringValidator;
+import org.jboss.tools.common.ui.databinding.ValueBindingBuilder;
+import org.jboss.tools.openshift.common.core.utils.VariablesHelper;
+import org.jboss.tools.openshift.internal.common.ui.utils.UIUtils;
+import org.jboss.tools.openshift.internal.ui.OpenShiftUIActivator;
+import org.jboss.tools.openshift.internal.ui.wizard.common.AbstractProjectPage;
+
+import com.openshift.restclient.OpenShiftException;
+
+/**
+ * @author Jeff Maury
+ *
+ */
+public class ResourcePayloadPage extends AbstractProjectPage<IResourcePayloadPageModel> {
+    
+    private static final IValidator URL_VALIDATOR =  new SimpleUrlStringValidator();
+
+    public ResourcePayloadPage(IWizard wizard, IResourcePayloadPageModel model) {
+        super(wizard, model,
+              "Select resource payload",
+              "Select the file or workspace file from where the resource will be created",
+              "resourcePayload");
+    }
+
+    @Override
+    protected void doCreateControls(Composite parent, DataBindingContext dbc) {
+        super.doCreateControls(parent, dbc);
+        createPayloadSourceControls(parent, dbc);
+    }
+
+    private void createPayloadSourceControls(Composite parent, DataBindingContext dbc) {
+        Group sourceGroup = new Group(parent, SWT.NONE);
+        sourceGroup.setText("Source");
+        GridDataFactory.fillDefaults()
+                .align(SWT.FILL, SWT.FILL)
+                .grab(true, true)
+                .span(3, 1)
+                .hint(SWT.DEFAULT, SWT.DEFAULT)
+                .applyTo(sourceGroup);
+        GridLayoutFactory.fillDefaults()
+                .numColumns(3)
+                .margins(10, 6)
+                .spacing(6, 6)
+                .applyTo(sourceGroup);
+        
+        createLocalSourceControls(dbc, sourceGroup);
+    }
+
+    private void createLocalSourceControls(DataBindingContext dbc, Group sourceGroup) {
+        Label label = new Label(sourceGroup, SWT.NONE);
+        GridDataFactory.fillDefaults().span(3, 1).applyTo(label);
+        label.setText("Enter a file path (workspace or local) or a full URL");
```

Comment:
I believe that we should have this label ending with a ":" like a widget labels in Eclipse.